### PR TITLE
Add CPU model name to trace files and traceRecord

### PR DESCRIPTION
**Pull request by skrakau**
This adds the CPU model name to the trace files (to allow using this information for a CO2 footprint plugin) and accordingly to the `traceRecord`.

It is not added to any tests yet.

### CODE REVIEW

**Comment by bentsherman** [comment]
Wondering if we should integrate this with the `arch` directive somehow. See #3786

File: modules/nextflow/src/main/groovy/nextflow/trace/TraceRecord.groovy [hunk 1]
```diff
@@ -87,6 +87,7 @@ class TraceRecord implements Serializable {
             syscw:      'num',      // -- /proc/$pid/io
             read_bytes: 'mem',      // -- /proc/$pid/io
             write_bytes:'mem',      // -- /proc/$pid/io
+            cpu_model:  'str',      // -- /proc/cpuinfo field 'model name'
```
Comment (pditommaso) [inline modules/nextflow/src/main/groovy/nextflow/trace/TraceRecord.groovy:90]: Let's add this at the bottom of the list, after `hostname`

**Comment by pditommaso** [comment]
I would expect the `arch` to be included in the CPU model

**Review by pditommaso — CHANGES_REQUESTED** [review]
This looks ok, but there's a failing test. Likey due to new field

**Comment by skrakau** [comment]
Hi, 
I moved the new `cpu_model` field to the end of the list, adjusted the `test-bash-wrapper-with-trace.txt` file to the new `command-trace.txt` since this caused an error in the `BashWrapperBuilderTest` test, and added an entry to the docs. I hope I didn't forget anything.

File: docs/tracing.md [hunk 2]
```diff
@@ -307,6 +307,9 @@ The following table shows the fields that can be included in the execution repor
   :::
 : The host on which the task was executed. Supported only for the Kubernetes executor yet. Activate with `k8s.fetchNodeName = true` in the Nextflow config file.
 
+`cpu_model`
+: The name of the CPU model which was used to execute the task. This data is read from file `/proc/cpuinfo`.
```
Comment (pditommaso) [inline docs/tracing.md:311]:
```suggestion
`cpu_model`
: :::{versionadded} 22.07.0-edge
  :::
: The name of the CPU model used to execute the task. This data is read from file `/proc/cpuinfo`.
```

File: modules/nextflow/src/main/groovy/nextflow/trace/TraceRecord.groovy [hunk 3]
```diff
@@ -100,7 +100,8 @@ class TraceRecord implements Serializable {
             error_action:'str',
             vol_ctxt: 'num',
             inv_ctxt: 'num',
-            hostname: 'str'
+            hostname: 'str',
+            cpu_model:  'str',      // -- /proc/cpuinfo field 'model name'
```
Comment (pditommaso) [inline modules/nextflow/src/main/groovy/nextflow/trace/TraceRecord.groovy:104]:
```suggestion
            cpu_model:  'str'
```

**Review by pditommaso — APPROVED** [review]

**Comment by pditommaso** [comment]
We are ready to go! Thanks for this useful contribution!

**Comment by skrakau** [comment]
Thanks for adding it!

**Pull request merged**